File: codeforces/19.11.29.3.java
import java.util.Scanner;

public class Main
{
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int m = scan.nextInt();
		for (int i = 0; i < m; i++) {
		    getAns(scan.nextInt());
		}
	}
	
	public static void getAns(int n) {
		//try again
	    int k = (int) (Math.sqrt(n));
	    
	    int asdf = k;
	    int total = k;
	    if (asdf == n/asdf) {
	        asdf -= 1;
	    }
	    total += asdf + 1;
        System.out.println(total);

	    
	    System.out.print("0");
	    
	    for (int i = 1; i < k + 1; i++) {
	        System.out.print(" " + i);
	    }
	    
	    if (k == n/k) {
	        k -= 1;
	    }
	    
	    for (int j = k; j > 0; j--) {
	        System.out.print(" " + n/j);
	    }
	    
	    System.out.println();
	}
}

File: codeforces/19.11.29.3.py
import math
m = int(input())

def getAns(n):
    k = math.floor(math.sqrt(n))
    string = "0"
    for i in range(1, k + 1):
        string += " " + str(i)
    total = k
    
    if (k == n//k):
        k -= 1
    
    for i in range(k, 0, -1):
        string += " " + str(n//i)
    
    total += k + 1
    
    print(total)
    print(string)
    
    

for i in range(m):
    getAns(int(input()))